src/Mastermind/Mastermind.ts

```diff
@@ -0,0 +1,10 @@
+import { CodeColours, HintColours } from '../constants/constants'
+
+export default class Mastermind {
+
+  secretCode: string[] = null;
+
+  constructor() {
+    this.secretCode = [CodeColours.RED, CodeColours.BLUE, CodeColours.GREEN, CodeColours.YELLOW]
+  } 
```

Comment:
Hum, that's good.
I was thinking of going through the "the user should provide the code to the `Mastermind`" instead of having the class itself generating it. I think that it would be simpler to test and get moving.
The main reason is that having the `Mastermind` generating the code by itself would result on tests that are hard to write and predict (they would basically be the code that plays the game I reckon.)
What are your thoughts on this?

To exemplify, I would have something like:
```suggestion
  constructor(code) {
    this.secretCode = code;
  } 
```

Or even better, to give them a red test to start we can leave the constructor empty and have the test code passing the code into it.
```suggestion
  constructor() {
  } 
```

Reply:
I really like the second idea, leave an empty constructor and a red test case so they can start off by making it green and go from there.